### ISSUE TITLE
AuthenticateAsync failes in Internet Explorer

### ISSUE THREAD

**Issue by Tobias-08** [DuendeSoftware/Support#738]
Hi,

we did a PoC based on the quickstart 2 sample.

Our PoC runs fine in Edge but fails in Internet Explorer during `AuthenticateAsync` in the `Callback` page ([Line46](https://github.com/DuendeSoftware/Samples/blob/281b22419ab7fafdf68ff60716eed33f2f56d267/IdentityServer/v6/Quickstarts/2_InteractiveAspNetCore/src/IdentityServer/Pages/ExternalLogin/Callback.cshtml.cs#L46)). We need to support Internet Explorer because of MS-OFBA based login initiated from Microsoft Office (using Internet Explorer).

Let me be clear that the original quickstart 2 runs in Internet Explorer, too.

Our main modification compared to the sample is introducing another IDSV-instance ("trust chain": client ->idsv ->idsv ->external provider). We also had to increase `maxQueryString` via `web.config` because our scenario produced query strings above 2000 characters. Also the transfered cookies seem to be quite large (chunked into 3 pieces).

I know that this issue might be specific to my scenario but does anyone have a hint why Internet Explorer might fail during `AuthenticateAsync` or how to debug further?

Thanks!
Tobias

**Comment by josephdecock** [comment]
The cookie size issue and oidc state data formatter cache are the first two things I would have suggested. I would try to look at the network requests and verify that the external authentication cookie is actually being sent by Internet Explorer. Maybe it is confused about the SameSite cookie attribute or something similar. But if authenticate async is failing to authenticate, that usually means you aren't sending the cookie.

**Comment by Tobias-08** [comment]
Hi @josephdecock,

thanks for your reply. I got one step further:

I realized that I could simplify my setup and remove one idsv instance. The auth chain is now `client ->idsv ->external provider`.

This works in Edge as expected but still does not work in Internet Explorer (MS-OFBA scenario). This time there isn't a second call to `ExternalLogin/Callback` but the resulting redirect to `/connect/authorize/callback` still fails (it is just not executed by IE; Fiddler-trace stops after the 302-result).

Now the interesting part regarding identity server:

In the above mentioned chain I configured idsv via `IdentityProviderRestrictions` to directly pass through the client to the external provider (because there is only this provider for this client; the idsv offers other providers for other clients). When this **restriction**/pass through is configured the **redirect fails** in IE. But when I **remove the restriction** and explicitly click on the external provider in the idsv's login page the **redirect works** in IE!

Do you have any thoughts why/how the `IdentityProviderRestrictions`-implementation could cause this? Or am I missing some additional setup if I use restrictions?

Thanks!

**Comment by brockallen** [comment]
When your callback page logs the user in are you setting the "idp" claim?

https://docs.duendesoftware.com/identityserver/v6/ui/login/session/#well-known-claims-issued-from-the-login-page

**Comment by Tobias-08** [comment]
Yes, we are setting the `idp`-claim (via `IdentityServerUser.IdentityProvider`).

**Comment by josephdecock** [comment]
I'm not sure what could be causing this in IE. A possible work-around would be to try to check if the browser is IE. If it is, then you show the form that the user would click on. You could even have some IE specific javascript that would autosubmit the form in IE only.

**Comment by Tobias-08** [comment]
@josephdecock Thanks! We'll try a workaround like you described.
